### PR TITLE
feat(hacbs-910): Create catalog directory

### PR DESCRIPTION
**Pull request by parasense**
Convert definitions to Tekton catalog.

This ports existing YAML definitions.
Hello-* tasks were excluded.
Milestone 5 & 6 pipelines were combined.

Signed-off-by: Jon Disnard <jdisnard@redhat.com>

### CODE REVIEW

**Comment by parasense** [comment]
@johnbieren @theflockers @davidmogar @scoheb 

PTAL

This PR creates the `catalog` sub-directory, and that's it. 
It's not connected with any CI yet.

**Comment by davidmogar** [comment]
As an early review I’d say that having something named *pipeline inside the pipeline directory is a bit redundant. 

Also, I’d say that it’s better to have tasks and pipelines directories instead of task and pipeline (singular).

**Comment by parasense** [comment]
> As an early review I’d say that having something named *pipeline inside the pipeline directory is a bit redundant.
Sure, yeah agreed that is a redundancy... I'll rename that to just "release".

> Also, I’d say that it’s better to have tasks and pipelines directories instead of task and pipeline (singular).
I would agree, and thought it was strange how Tekton Catalog uses the singular instead of the plural. Please see the document on why it's done this way: 

https://github.com/tektoncd/community/blob/main/teps/0003-tekton-catalog-organization.md

TLDR: In short, the directory is named after the Tekton resource type, E.G. "pipeline" or "task", etc... which are regrettably defined in the singular form.

**Comment by johnbieren** [comment]
> > As an early review I’d say that having something named *pipeline inside the pipeline directory is a bit redundant.
> 
> Sure, yeah agreed that is a redundancy... I'll rename that to just "release".
I actually disagree with David on this and think the name would be nice as you seemingly originally had it. But I am not strongly opinionated on it and am fine with you changing it.
> 
> > Also, I’d say that it’s better to have tasks and pipelines directories instead of task and pipeline (singular).
> 
> I would agree, and thought it was strange how Tekton Catalog uses the singular instead of the plural. Please see the document on why it's done this way:
> 
> https://github.com/tektoncd/community/blob/main/teps/0003-tekton-catalog-organization.md
> 
> TLDR: In short, the directory is named after the Tekton resource type, E.G. "pipeline" or "task", etc... which are regrettably defined in the singular form.

I really wish GitHub was a little smarter on the files changed view. None of this is new code so far, right? Just moved to different directories and filenames?

**Comment by parasense** [comment]
> > > As an early review I’d say that having something named *pipeline inside the pipeline directory is a bit redundant.
> > 
> > 
> > Sure, yeah agreed that is a redundancy... I'll rename that to just "release".
> 
> I actually disagree with David on this and think the name would be nice as you seemingly originally had it. But I am not strongly opinionated on it and am fine with you changing it.

Same; I have no firm opinion here, so I say we just leave it the way it was changed unless there is a super compelling reason to say `release-pipeline` how it was originally? On that note, I have to say thanks! You made me take a second look, and I noticed an error. I had it looking like this
```
release/
├── 0.1
│   ├── README.md
│   ├── release-pipeline.yaml  # Mistake, should be "release.yaml"
│   ├── samples
│   └── tests
├── 0.2
│   ├── README.md
│   ├── release-pipeline.yaml  # Mistake, should be "release.yaml"
│   ├── samples
│   └── tests
├── OWNERS
└── README.md
```
The the YAML definition file must be the same name as the directory. So if we were to rename `release-pipeline` to `release`, then the YAML definitions must change accordingly, and apparently I forgot that step, so I got another patch incoming for that.

[snip]

> I really wish GitHub was a little smarter on the files changed view. None of this is new code so far, right? Just moved to different directories and filenames?

Correct, all the YAML files are from the existing, and agreed about GitHub, it's not great seeing the flood of duplicate files. I'm sorry for that. :cry: 

I may have renamed some strings inside the YAML for `metadata.name`, for example when changing the above YAML filenames, the string inside was changed:
```
$ yq '.metadata.name' release.yaml 
release
```
That used to say "release-pipeline". 
If this change might potentially cause a backwards compatibility issue, we can probably use that as the "compelling reason" to justify going back to how it was.

There are a few totally new `README.md` files, for example where none existed they were created. I used your `README.md`'s as the template when creating those, like the table columns. 

One interesting note....
The existing `README.md`'s can exist at two levels, at the top level of the resource, and down in the version sub-directories. So I went ahead and duplicated the `README.md` in those two places, which really amplified the size of this PR.
See here:
https://github.com/tektoncd/community/blob/main/teps/0003-tekton-catalog-organization.md#organization
(grep for "Structure of the task")

File: catalog/pipeline/release/0.2/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Milestone-6 Release Pipeline
+
+Tekton pipeline to release HACBS Application Snapshot.
```
Comment (johnbieren) [inline catalog/pipeline/release/0.2/README.md:3]: can we add `to Quay` to the end of this string too?

File: catalog/pipeline/release/0.2/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+| Name | Description | Optional | Default value |
+|------|-------------|----------|---------------|
+| applicationSnapshot | The ApplicationSnapshot in JSON format | No | - |
+| policy | Name of the policy to apply when validating the enterprise contract | No | - |
+| extraConfigGitUrl |URL to the remote Git repository containing the extra config | No | - |
+| extraConfigGitRevision | Revision to fetch from the remote Git repository containing the extra config | No | - |
+| extraConfigPath | Path to the extra config file within the repository | No | - |
+| postCleanUp | Cleans up workspace after finishing executing the pipeline | Yes | true |
+
+## Changes since 0.1 (milestone-5)
```
Comment (johnbieren) [inline catalog/pipeline/release/0.2/README.md:16]: this is nice

File: catalog/task/apply-mapping/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# apply-mapping
```
Comment (johnbieren) [inline catalog/task/apply-mapping/README.md:1]: I'm not sure I like having duplicate READMEs in both the version directory and the top level directory. I really like how you have the pipeline directory setup, so I suppose I am in favor of READMEs in the version directories but not top level directories. Maybe if we still want READMEs in both spots the top level one would be far less detailed... but that may be a little awkward to construct at least for the pipelines

File: catalog/task/cleanup-workspace/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# cleanup-workspace
```
Comment (johnbieren) [inline catalog/task/cleanup-workspace/README.md:1]: same comment as the apply-mapping task readme

File: catalog/task/prepare-validation/README.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
+# prepare-validation
```
Comment (johnbieren) [inline catalog/task/prepare-validation/README.md:1]: same comment as the apply-mapping readme

File: catalog/task/push-application-snapshot/README.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# push-application-snapshot
```
Comment (johnbieren) [inline catalog/task/push-application-snapshot/README.md:1]: same comment as the apply-mapping readme

File: catalog/task/skopeo-copy/README.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
+# push-application-snapshot
```
Comment (johnbieren) [inline catalog/task/skopeo-copy/README.md:1]: same comment as the apply-mapping readme

**Review by johnbieren — CHANGES_REQUESTED** [review]
The skopeo-copy task directory seems to have an empty OWNERS file. Is that intended? Also, will the CI be addressed in a different PR?

File: catalog/task/apply-mapping/README.md (hunk 3 above)
Comment (parasense) [reply catalog/task/apply-mapping/README.md]: Thanks! 
I'm the top-level README is "optional", where the versioned README is "recommended". So I removed the top-level READMEs.

File: README.md [hunk 8]
```diff
@@ -10,13 +10,14 @@ All the bundles in this repository are pushed to our [Quay organization](https:/
...
 
+* **version**: This tag will point Catalog version of the Tekton resource, for example "0.1", 0.2", etc..
 * **main**: This tag will point always to the existing code in the repository main branch.
 * **<commit-sha>**: A tag with the commit-sha of the change will be created, so we can have multiple bundles per each
 change in the git repository.
 
 ## Adding new bundles
 
-To add new bundles to this repository, create a directory in the [definitions directory](definitions) and place inside
+To add new bundles to this repository, create a directory in the [catalog directory](catalog) and place inside
```
[the remark this answers is not in the document]
Comment (parasense) [reply README.md]: I've added some words, but I'm not sure it makes sense. It could probably be more or clear.
I'm tempted to show a link to the Tekton Catalog docs:
https://github.com/tektoncd/community/blob/main/teps/0003-tekton-catalog-organization.md#organization

File: README.md [hunk 9]
```diff
@@ -10,13 +10,14 @@ All the bundles in this repository are pushed to our [Quay organization](https:/
 This push happens automatically whenever a new bundle is added to the repository or an existent one is updated. Two new
 tags will be created or updated during this automatic process:
 
+* **version**: This tag will point Catalog version of the Tekton resource, for example "0.1", 0.2", etc..
```
[the remark this answers is not in the document]
Comment (parasense) [reply README.md]: It was a typo, does it look better now?

**Comment by parasense** [comment]
> The skopeo-copy task directory seems to have an empty OWNERS file. Is that intended? Also, will the CI be addressed in a different PR?

That was a good catch! Apparently I missed that file. But it should be fixed now.

File: README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (johnbieren) [reply README.md]: I think it is great how it is now

File: catalog/pipeline/release/0.1/README.md [hunk 10]
```diff
@@ -0,0 +1,14 @@
+# Milestone-5 Release Pipeline
+
+Tekton pipeline to release HACBS Application Snapshot.
```
[the remark this answers is not in the document]
Comment (johnbieren) [reply catalog/pipeline/release/0.1/README.md:3]: This doesn't seem to be fixed, but everything else is. I am happy to approve once this line is

[the remark this answers is not in the document]
Comment (parasense) [reply catalog/pipeline/release/0.1/README.md]: Somehow that one slipped through.
Thanks for catching the mistake! :+1:

**Review by johnbieren — APPROVED** [review]
